platform/resource-bundle-locator/src/main/java/ddf/platform/resource/bundle/locator/ResourceBundleLocatorImpl.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.platform.resource.bundle.locator;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Locale;
+import java.util.MissingResourceException;
+import java.util.ResourceBundle;
+
+public class ResourceBundleLocatorImpl implements ResourceBundleLocator {
+
+  private File resourceBundleBaseDir;
+
+  @Override
+  public ResourceBundle getBundle(String baseName) {
+    return getBundle(baseName, Locale.getDefault());
+  }
+
+  @Override
+  public ResourceBundle getBundle(String baseName, Locale locale) {
+    File resourceBundleDir = resourceBundleBaseDir;
+    if (resourceBundleDir == null) {
+      resourceBundleDir = new File(String.format("%s/etc/i18n/", System.getProperty("ddf.home")));
+    }
+
+    try {
+      URL[] urls = {resourceBundleDir.toURI().toURL()};
+
+      try (URLClassLoader loader = new URLClassLoader(urls)) {
+        return AccessController.doPrivileged(
+            (PrivilegedAction<ResourceBundle>)
+                () -> ResourceBundle.getBundle(baseName, locale, loader));
+      }
+    } catch (IOException e) {
```

Comment:
❗️ I am questioning not throwing back the IOException. By design this API is to load resources bundles on behalf of the client. The client should deal with these exceptions. Hiding them inside a MissingResourceException when in fact it ain't related at all to that makes it weird and confusing to the users of this API. It is like saying when you read a file that we wrap all IO errors as file not found wether it is there or not.

Reply:
I agree in this case. I think we're hiding the actual error by wrapping it in a `MissingResourceException`. I'm just going to log the error here instead

Reply:
My point is that we shouldn't be hiding it. We should declare that we can throw back IOException

Reply:
Oh gotcha I understand. Yeah I can make that change